module-4/hydra/hydra.py

```diff
@@ -1 +1,82 @@
-print("Initializing Hydra")
\ No newline at end of file
+import argparse
+import paramiko
+import itertools
+import string
+import time
+import signal
+import sys
+
+def signal_handler(sig, frame):
+    print('\n[!] Signal received. Exiting gracefully.')
+    sys.exit(0)
+
+signal.signal(signal.SIGINT, signal_handler)  # Handle Ctrl+C (SIGINT)
+
+def brute_force_ssh(username, server, charset, min_length, max_length):
+    client = paramiko.SSHClient()
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+    for length in range(min_length, max_length + 1):
+        for password in itertools.product(charset, repeat=length):
+            password = ''.join(password)
+            print(password)
```

Comment:
in production code we do not want to print something without a reason or real logging format.
Remove.

When refactored you may print something similar when trying to really use this password for an attack, like follows:

```python
...
pw_list = [] # generated beforehand
for pw in pw_list:
    try:
        ssh_connect(pw, IP)
    except Exception as e:
        # handle error
# ...
```

Reply:
you did not change a thing here, right? Why resolve the thread then?